media/js/app/taxonomy/taxonomy.js

```diff
@@ -486,110 +488,42 @@
             var self = this;
 
             jQuery.get(onomyURL, function(data) {
-                var x = data;
-
                 var parents = [];
```

Comment:
Could the parents array be modeled as a map, indexed by display_name? This would allow easy lookup and mitigate the need for the inline search.